patientsearch/src/js/components/TimeoutModal.js

```diff
@@ -53,14 +53,22 @@ export default function TimeoutModal() {
         let tokenData = null;
         try {
           tokenData = JSON.parse(response);
+          let accessTokenExpiresIn = parseFloat(tokenData["access_expires_in"]);
+          let refreshTokenExpiresIn = parseFloat(tokenData["refresh_expires_in"]);
+          let refreshTokenOnVentilator = refreshTokenExpiresIn < accessTokenExpiresIn;
           //in seconds
-          expiresIn = tokenData["expires_in"];
+          //1. check if refresh token will expire before access token first
+          //2. check if access token will expire
+          expiresIn = refreshTokenOnVentilator ? refreshTokenExpiresIn : accessTokenExpiresIn;
+          let tokenAboutToExpire = Math.floor(expiresIn) >= 1 && Math.floor(expiresIn) <= 60;
+          //flag for whether to prompt the user to refresh session, i.e. request another access token
+          refresh = tokenAboutToExpire && !refreshTokenOnVentilator ? true: false;
 
           if (!tokenData["valid"] || expiresIn <= 1) {
             handleLogout();
             return;
           }
```

Comment:
I tested again and noticed the frontend redirecting to `/logout` when the Access Token expired, when it should only redirect to `/logout` if `refreshTokenOnVentilator`, ie we will not be able to get a new Access Token from Keycloak.

```suggestion
          if (refreshTokenOnVentilator && (!tokenData["valid"] || expiresIn <= 1)) {
            handleLogout();
            return;
          }
```


Reading through the rest of the code in this file, I also noticed[ `handleLogout();` being called when the backend responds with a 401](https://github.com/uwcirg/cosri-patientsearch/pull/39/files#diff-a01a2145351da25a7b912bb1b574c1655bdebcd5adb9156eb0a1cf31a23ef6b7R86-R90), when we should instead redirect to `/`, to trigger the backend to request a new Access Token, or prompt a user to log in again (if their Keycloak session has expired).

Reply:
Reading through the rest of the code in this file, I also noticed[ `handleLogout();` being called when the backend responds with a 401](https://github.com/uwcirg/cosri-patientsearch/pull/39/files#diff-a01a2145351da25a7b912bb1b574c1655bdebcd5adb9156eb0a1cf31a23ef6b7R86-R90), when we should instead redirect to `/`, to trigger the backend to request a new Access Token, or prompt a user to log in again (if their Keycloak session has expired).

Reply:
thank you @ivan-c I did just make the fix.